### PR TITLE
MAGECLOUD-3546: Rewrite PostDeployTest to use Cloud Docker

### PR DESCRIPTION
**Pull request by BaDos**
### Description

Rewrite PostDeployTest to use Cloud Docker

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3546

### Manual testing scenarios

N/A

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 dist: trusty
-sudo: required
+sudo:
```
Comment (shiftedreality) [inline .travis.yml]: is this entry needed?

Comment (bbatsche) [inline .travis.yml]: Should there be a value here?

**Review by bbatsche — CHANGES_REQUESTED** [review]

File: tests/functional/_data/files/.magento.env.yaml.scdondemand [hunk 2]
```diff
@@ -0,0 +1,3 @@
+stage:
```
Comment (shiftedreality) [inline tests/functional/_data/files/.magento.env.yaml.scdondemand]: Leave original extension of files to have it correctly parsed in IDE. Let's rename to `scdondemand.magento/env.yaml`

**Review by shiftedreality — DISMISSED** [review]

File: .travis.yml (hunk 1 above)
Comment (BaDos) [reply .travis.yml]: I had some conflicts during merging.
Fixed.

Comment (BaDos) [reply .travis.yml]: See my comment above.

File: tests/functional/_data/files/.magento.env.yaml.scdondemand (hunk 2 above)
Comment (BaDos) [reply tests/functional/_data/files/.magento.env.yaml.scdondemand]: Sure.

File: src/Test/Functional/Acceptance/PostDeployCest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+//    /**
+//     * @return array
+//     */
+//    public function postDeployDataProvider(): array
+//    {
+//        return [
+//            ['variables' => ['ADMIN_EMAIL' => 'admin@example.com']],
+//            ['variables' => []],
+//        ];
+//    }
```
Comment (bbatsche) [inline src/Test/Functional/Acceptance/PostDeployCest.php]: Commented code?

**Review by bbatsche — CHANGES_REQUESTED** [review]

File: src/Test/Functional/Acceptance/PostDeployCest.php (hunk 3 above)
Comment (BaDos) [reply src/Test/Functional/Acceptance/PostDeployCest.php]: 🙈

**Review by bbatsche — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Review by billygilbert — APPROVED** [review]

**Pull request merged**